### ISSUE TITLE
Request to Add Language Switching Options in the Top-Right Corner of the Topics Page

### ISSUE THREAD

**Issue by bittomhan** [bitcoinops/bitcoinops.github.io#1273]
# Request to Add Language Switching Options in the Top-Right Corner of the Topics Page

## Current Issues:

1. When adding translated content directly, the Chinese content is sorted according to its first character and displayed within the English alphabet listing.
2. Currently, when Chinese content appears in English folders, two versions are displayed together. This issue can be addressed when the content is distinguished on subsequent pages, so it's not a concern for now.

## Current Approach:

1. There's a need to create a separate `ZH` page available from the top-right corner for users to select. The layout should refer to how `newsletter` or `publications` type pages render their content.
2. Given this, considering the `_layout` directory, `newsletter` has both `.md` and `.html` files, while `publications` only has `.md` files. The topic needing modification only has `.html` files, and a direct copy-paste modification won't work. We need to understand the logical relationship between these.
3. The layout for `topic.html` is `post.html`. Code changes should be modeled on `post.html`.
4. Add a similar `{% assign path = "/" | append: page.lang | append: "/topics/" %}` at the top, like in `newsletter`.
5. From what I observe, after adding the above line, the rendered page does not differentiate languages. I believe that the solution involves adding `lang` and `permalink` fields in the front matter (`schema`) of the English `.md` files under the `topics` folder. This should allow subsequent pages to render with language options in the top-right corner.

## Assistance Needed:

1. Standardize the schema setup to accommodate the addition of translated content in Chinese.
2. Guidance on how to set up language tags in the top-right corner within `post.html` or `topic.html`.
3. Hint: We may need to add language-handling content within the `topic.html` schema.

I don't have the technical expertise to solve these issues; your assistance is greatly appreciated.

**Comment by bittomhan** [comment]
Can you help me fix this? @bitschmidty

**Comment by harding** [comment]
Hi @bittomhan 

I don't have a link handy, but in previous discussion about topic page content, I've been somewhat concerned about translations of that.  Translating newsletters has the advantage that we don't expect newsletters to change after publication (except for small typo fixes).  However, topic pages can have their content updated frequently, not just with new items but also with revisions to the summary text and references.  If we begin to do translations of topic entries, we'll need to implement some sort of versioning system that ensures the non-English localized content is fundamentally similar to the English original content.

I'm certainly not opposed to localized topic content, but I just wanted to make you aware that it might be a larger project with additional maintenance burden than it might initially appear to be.

**Comment by bittomhan** [comment]
> Hi @bittomhan
> 
> I don't have a link handy, but in previous discussion about topic page content, I've been somewhat concerned about translations of that. Translating newsletters has the advantage that we don't expect newsletters to change after publication (except for small typo fixes). However, topic pages can have their content updated frequently, not just with new items but also with revisions to the summary text and references. If we begin to do translations of topic entries, we'll need to implement some sort of versioning system that ensures the non-English localized content is fundamentally similar to the English original content.
> 
> I'm certainly not opposed to localized topic content, but I just wanted to make you aware that it might be a larger project with additional maintenance burden than it might initially appear to be.

Hi David,

Thank you so much for your response.

I've uploaded some of my translated content to the link I mentioned earlier. I'm pleased to report that I'm over halfway done, and if things go as planned, I anticipate completing the translations within a month.

I'm glad to hear from you about the extensive effort required to translate Topics. Indeed, through these translations, I've gained a deeper understanding of Bitcoin's technical developments and progress. As an enthusiast genuinely interested in Bitcoin technology, diving into Topics to grasp the foundational concepts is my initial step. I will continue translating the current content, which will not only further my understanding of Bitcoin's technology and development but also serve as a foundation for future updates and improvements. I'm excited about the prospect of contributing to this community.

Additionally, I plan to gradually translate the Newsletter, especially the content spanning nearly five years. My goal is to weave together a clear narrative of the technical development and evolution. I aspire to explain the evolution of Bitcoin technology in Chinese, which I believe would be valuable for many looking to contribute to Bitcoin's technological content.

Thank you once again for your support and understanding.

Warm regards,
BitTom
